be/src/storage/lake/tablet_reshard.cpp

```diff
@@ -0,0 +1,369 @@
+        set_all_data_files_shared(&rowset_metadata);
+    }
+
+    // delvec
+    if (tablet_metadata->has_delvec_meta()) {
+        for (auto& pair : *tablet_metadata->mutable_delvec_meta()->mutable_version_to_file()) {
+            pair.second.set_shared(true);
+        }
+    }
+
+    // dcg
+    if (tablet_metadata->has_dcg_meta()) {
+        for (auto& dcg : *tablet_metadata->mutable_dcg_meta()->mutable_dcgs()) {
+            auto* shared_files = dcg.second.mutable_shared_files();
+            shared_files->Clear();
+            shared_files->Resize(dcg.second.column_files_size(), true);
+        }
+    }
+
+    // sst
+    if (tablet_metadata->has_sstable_meta()) {
+        for (auto& sstable : *tablet_metadata->mutable_sstable_meta()->mutable_sstables()) {
+            sstable.set_shared(true);
+        }
+    }
+}
+
+static Status handle_splitting_tablet(TabletManager* tablet_manager, const SplittingTabletInfoPB& splitting_tablet,
+                                      int64_t base_version, int64_t new_version, const TxnInfoPB& txn_info,
+                                      std::unordered_map<int64_t, TabletMetadataPtr>& new_metadatas,
+                                      std::unordered_map<int64_t, TabletRangePB>& tablet_ranges) {
+    // Check tablet metadata cache
+    {
+        auto old_tablet_new_metadata_location =
+                tablet_manager->tablet_metadata_location(splitting_tablet.old_tablet_id(), new_version);
+        auto cached_old_tablet_new_metadata =
+                tablet_manager->metacache()->lookup_tablet_metadata(old_tablet_new_metadata_location);
+        if (cached_old_tablet_new_metadata == nullptr) {
+            goto CONTINUE_HANDLE_SPLITTING_TABLET;
+        }
+
+        new_metadatas.emplace(splitting_tablet.old_tablet_id(), std::move(cached_old_tablet_new_metadata));
+
+        for (auto new_tablet_id : splitting_tablet.new_tablet_ids()) {
+            auto new_tablet_new_metadata_location =
+                    tablet_manager->tablet_metadata_location(new_tablet_id, new_version);
+            auto cached_new_tablet_new_metadata =
+                    tablet_manager->metacache()->lookup_tablet_metadata(new_tablet_new_metadata_location);
+            if (cached_new_tablet_new_metadata == nullptr) {
+                new_metadatas.clear();
+                tablet_ranges.clear();
+                goto CONTINUE_HANDLE_SPLITTING_TABLET;
+            }
+
+            new_metadatas.emplace(new_tablet_id, std::move(cached_new_tablet_new_metadata));
+            tablet_ranges.emplace(new_tablet_id, TabletRangePB());
+        }
+
+        // All new metadatas found in cache, return ok
+        return Status::OK();
+    }
+
+CONTINUE_HANDLE_SPLITTING_TABLET:
+
+    auto old_tablet_old_metadata_or =
+            tablet_manager->get_tablet_metadata(splitting_tablet.old_tablet_id(), base_version, false);
+    if (old_tablet_old_metadata_or.status().is_not_found()) {
```

Comment:
Is that means the split task has been finished, and this is a redundant RPC call for it?